evm-gateway/src/EVMGateway.ts

```diff
@@ -48,6 +50,17 @@ function memoize<T>(fn: () => Promise<T>): () => Promise<T> {
   };
 }
 
+// traverse mapping at slot using key solidity-style
+// https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html#mappings-and-dynamic-arrays
+function followSolidityMapping(slot: bigint, key: string) {
+  return BigInt(solidityPackedKeccak256(['bytes', 'uint256'], [key, slot]));
+}
+
+// see: EVMProofHelper.uint256FromBytes()
+function uint256FromBytes(hex: string): bigint {
+  return BigInt(hex.slice(0, 66));
```

Comment:
Is this slice necessary? Length check instead.

Reply:
The purpose is to mimic the [Solidity function](https://github.com/unruggable-labs/evmgateway-v2/blob/edf5465616c98e3d3bcb6319efafe2ae6aea5c00/evm-verifier/contracts/EVMProofHelper.sol#L23) which accepts any length of bytes.  

In Solidity, bytes32 of more than 32 bytes is the leading 32 bytes.

Actually, looking at the /v2/ code, this is missing the special case for `BigInt('0x')` which is a SyntaxError.